mozjs/js/rust/tests/magic_dom.rs

```diff
@@ -92,10 +92,10 @@ fn test_DOMPoint_magic_layout() {
     // the `*mut jsapi::JSObject`.
     let instance: DOMPoint = unsafe { mem::zeroed() };
     assert_eq!(&instance as *const _ as usize, &instance.object as *const _ as usize);
-    assert_eq!(&instance.x as *const _ as usize, &instance.object as *const _ as usize);
-    assert_eq!(&instance.y as *const _ as usize, &instance.object as *const _ as usize);
-    assert_eq!(&instance.z as *const _ as usize, &instance.object as *const _ as usize);
-    assert_eq!(&instance.w as *const _ as usize, &instance.object as *const _ as usize);
+    assert_eq!(&instance.x as *const _ as usize - 8, &instance.object as *const _ as usize);
+    assert_eq!(&instance.y as *const _ as usize - 8, &instance.object as *const _ as usize);
+    assert_eq!(&instance.z as *const _ as usize - 8, &instance.object as *const _ as usize);
+    assert_eq!(&instance.w as *const _ as usize - 8, &instance.object as *const _ as usize);
```

Comment:
\o/ tests passing!

However, we also need to update `js::magic::MagicSlot::get_object` to not be off by a word, too.

I expect on 32 bits, `8` is wrong, and what we really need here is to remove one word:

```rust
&instance.x as *const _ as usize - mem::size_of<usize>()
```

Luckily if we ever start trying to use this code on 32bit, this test will start failing loudly and let us know if we're doing things wrong.